helix-term/src/commands.rs

```diff
@@ -1816,9 +1816,13 @@ mod cmd {
         let jobs = &mut cx.jobs;
         let (_, doc) = current!(cx.editor);
 
+        // Should refresh lang server or not
+        let mut refresh_ls = false;
+
         if let Some(path) = path {
             doc.set_path(Some(path.as_ref()))
                 .context("invalid filepath")?;
+            refresh_ls = true; // we want to refresh the lang server
```

Comment:
I think we only need to `refresh_ls` when we save from scratch to a new path, we can check `doc.path().is_none()` first.

Reply:
Yeah, we could also extend this by checking the old language before save, and new language after save, then emitting a `textDocument/close` event to the old LSP and starting a new LSP if the filetype uses a different language server.

Reply:
> Yeah, we could also extend this by checking the old language before save, and new language after save, then emitting a `textDocument/close` event to the old LSP and starting a new LSP if the filetype uses a different language server.

sounds like a good idea, can you give me some pointers for that tho?

Reply:
```rust
let old_language_server = doc.language_server.clone();

// ...

if old_language_server != doc.language_server() {
        if let Some(language_server) = old_language_server {
            tokio::spawn(language_server.text_document_did_close(doc.identifier()));
        }

       if let Some(......) = doc.language_server() {
          // basically that whole block you have under refresh
       }
}
```